### ISSUE TITLE
Investigate exp()

### ISSUE THREAD

**Issue by pmur** [libdfp/libdfp#22]
decimal32 and decimal64 have high ulps (15 and 13 respectively) with the input 88.72269439697265625.

**Comment by pmur** [comment]
D64 and D32 utilize an approximation which strips off fraction off the exponent. This causes some loss of precision, and therefore some carrying error when summing the taylor series. D128 falls back to libdecnumber, which claims <= 1 ulp.

In the case above,  there is a bit of drift in the summing of the taylor series. The sum has 8 ulp of error, which is further compounded by the multiplication of the tabular exp(88).

In short, these functions are trading accuracy for speed.

**Comment by tuliom** [comment]
Does it make sense to add a comment to the source code saying that?

**Comment by pmur** [comment]
... And this was a bug in the test code as both the input and output values were rounded to produce a final value.

The rounded input needs to be passed through the oracle to avoid cases where the epsilon between inputs produces more than 1ULP of difference in the result.
